cypress-io/cypress
# feat: distinguish app vs launchpad utm_source when using utm params

Closes https://cypress-io.atlassian.net/browse/UNIFY-1436

### User facing changelog

No user facing changes, but onlinks that previously had no `utm_source`, now correctly set a source of 'Binary: App' and 'Binary: Launchpad' depending on where they were clicked.

### Additional details

In 9.x the `utm_source` query param of `Test Runner` was added on the server as a part of `openExternal`. This PR removes that functionality completely from the server, and instead applies the same rule on the client side, in our `getUrlWithParams` helper function, to add the `utm_source` automatically to URLs that already include any other utm params.

Among other things, this makes the full length URL testable from the client side, and allows us to continue having well-formed links with accurate `href` attributes.

This PR covers all links in the docs menu and the growth prompt that's part of it, and any future links that use utm params will have the correct source appended automatically.


### How to test manually

Click any links in the docs menu in the launchpad and app and verify that when they take you to the docs pages, the utm params are present in the page URL and include the correct utm_source - `Binary: App` for the app and `Binary: Launchpad` for the browser.

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
